scripts/cleanup_openstack_instance_snapshot.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Cleanup unused instance snapshot from glance repository
+
+Usage: scripts/cleanup_openstack_instance_snapshot.py [optional list of provider keys]
+
+If no providers specified, it will cleanup all of them.
+
+"""
+import sys
+from datetime import datetime
+from datetime import timedelta
+
+import iso8601
+import tzlocal
+
+from cfme.utils.log import add_stdout_handler
+from cfme.utils.log import logger
+from cfme.utils.providers import get_mgmt
+from cfme.utils.providers import list_provider_keys
+
+LOCAL_TZ = tzlocal.get_localzone()
+GRACE_TIME = timedelta(hours=2)
+TIME_LIMIT = datetime.now(tz=LOCAL_TZ) - GRACE_TIME
+
+# log to stdout too
+add_stdout_handler(logger)
+
+
+def main(*providers):
+    """ Cleanup all snapshots name starting with test_snapshot and created by >= 2 hours before
+
+    :param providers: Lists provider keys
+    :return:
+    """
+
+    for provider_key in providers:
+        logger.info("Cleaning up {}".format(provider_key))
+        provider_obj = get_mgmt(provider_key)
+
+        try:
+            images = provider_obj.list_templates()
+        except Exception as e:
+            logger.exception("Connect to provider failed:{} {} {}".format(
+                provider_key, type(e).__name__, str(e)))
+            continue
+
+        for img in images:
+            if "test_snapshot_" in img.name \
```

Comment:
optional:
please avoid the use of `\`

Reply:
This can certainly be re-formatted to avoid.

Alternatively, this could be applied much earlier, when making the API query in the first place.

`images = [i for i in provider_obj.list_templates() if i.name.startswith('test_snapshot_')]`

Even better, expand the script to use ArgParse and take a CLI arg of what image names to remove, so that its not hardcoded as 'test_snapshot'. :)  But that's optional.

Reply:
Thank you for suggesting ArgParse idea, Implemented it.  For your other suggestion of using list comprehension , it will require looping 2 times, once as you mentioned above and next when i need to loop the images for deleting it. So i have keep it as it is and removed the concern "\" :)